### PR TITLE
Add lint check on ci

### PR DESCRIPTION
**Pull request by agustinhenze**
* Add pylint + pep8 on .travis.yml
* Make tests comply with pep8 requirements

### CODE REVIEW

**Comment by agustinhenze** [comment]
It closes #13

File: .pylintrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[FORMAT]
+# PEP8 compliant
+max-line-length=79
```
Comment (spbnick) [inline .pylintrc]: Ah, so PEP8 is stricter than Google Python Style Guide. Alright. However, we'll need to then fork the guidelines and adjust the column number in there. We should probably put the guidelines into https://github.com/RH-FMK/meta

File: tests/misc.py [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
 #
 # You should have received a copy of the GNU General Public License
 # along with this program; if not, write to the Free Software
-# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301,
+# USA.
```
Comment (spbnick) [inline tests/misc.py]: Could you please rewrap this paragraph as a whole, here and everywhere else, so we don't have an extra line with the hanging "USA"? It should fit.

File: .travis.yml [hunk 3]
```diff
@@ -3,4 +3,7 @@ python:
     - "2.7"
 install:
     - pip install requests junit_xml beaker-client mock
-script: python -m unittest discover tests
+    - pip install pylint
+script:
+    - python -m unittest discover tests
+    - pylint tests
```
Comment (spbnick) [inline .travis.yml]: Could you please put the pylint invocation first, as it's a more basic, lower level test?

**Comment by spbnick** [comment]
Oh, and please add a line saying "Fixes #12" to the message of the commit adding pylint to .travis.yml.

File: tests/misc.py (hunk 2 above)
Comment (agustinhenze) [reply tests/misc.py]: Ok, I have tried again and I get `C: 16, 0: Line too long (80/79) (line-too-long)` from pylint.

File: .travis.yml (hunk 3 above)
Comment (agustinhenze) [reply .travis.yml]: My reasoning was that I'd prefer to see what is the result of the test suite when a PR is opened and if pylint doesn't pass is not something useful for the review. It has to pass if you want to see the PR accepted but it does not contribute too much in the discussions.

File: .pylintrc (hunk 1 above)
Comment (agustinhenze) [reply .pylintrc]: Maybe we can mention that we follow the Google Python Style Guide but pylint is on top of it. If you get a pylint error it is something that you have to fix. With the idea in mind to maintain less dev doc :)

Comment (spbnick) [reply .pylintrc]: Yeah, we can have a "contributing" document refer to the style guide and have exceptions on top. That will work as long as exceptions are not numerous. We'll have to spell them out, though, not just point at the CI code.

File: tests/misc.py (hunk 2 above)
Comment (spbnick) [reply tests/misc.py]: I mean something like this:

    # You should have received a copy of the GNU General Public License along with
    # this program; if not, write to the Free Software Foundation, Inc., 51
    # Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.

File: .travis.yml (hunk 3 above)
Comment (spbnick) [reply .travis.yml]: I would just like to have some logical order, it doesn't matter which. I agree, having it "most important first" or "biggest impact first" makes sense, and will work. Could you perhaps put a comment above, specifying which order the tests should be, so we don't have to discuss this again? You can use `echo` to print it, or just `true` with comment as the argument, which will be ignored.

File: tests/misc.py (hunk 2 above)
Comment (agustinhenze) [reply tests/misc.py]: Ha! You're right, I misunderstood you before. Pushed

**Comment by spbnick** [comment]
I mean put "Fixes #12" line into the commit message *body* (at the end), not *subject*. Sorry if I was unclear. GitHub will then see it and close the issue automatically when I push the commit.

File: .travis.yml (hunk 3 above)
Comment (spbnick) [reply .travis.yml]: Ah, on the second thought, adding the comment about the order matters little at this moment, please feel free to ignore.

File: .pylintrc (hunk 1 above)
Comment (agustinhenze) [reply .pylintrc]: Ok, agreed

File: .travis.yml (hunk 3 above)
Comment (agustinhenze) [reply .travis.yml]: Ok

File: tests/test_reporter.py [hunk 4]
```diff
@@ -34,6 +35,7 @@ def request_get_mocked(filename):
             filename: Asset's filename.
         """
         get_mocked = mock.Mock()
+
```
Comment (spbnick) [inline tests/test_reporter.py]: This is a needed change, but it's not about complying with the line length requirements as commit subject says.
Please either make it a separate commit, or change the commit subject to something like "Make tests comply with pep8 requirements". While this is a small matter, it's easier to simply always keeping commit descriptions true, than deciding every time if it's important or not.

Comment (agustinhenze) [reply tests/test_reporter.py]: You are so right. I have completely forgot that I have been running pep8 locally... Please take a look to the last push, I have added pep8 and remove the .pylintrc file to keep it simple as possible.

Comment (agustinhenze) [reply tests/test_reporter.py]: Well I've changed it by pycodestyle because pep8 is deprecated as you can see [here](https://travis-ci.org/RH-FMK/skt/builds/353475444?utm_source=github_status&utm_medium=notification)

**Comment by spbnick** [comment]
Thanks a lot, Agustin, merged!
